frontend/src/components/layout/whatsnew/WhatsNewMenu.tsx

```diff
@@ -0,0 +1,303 @@
+import {
+  forwardRef,
+  HTMLAttributes,
+  ReactNode,
+  RefObject,
+  useRef,
+} from "react";
+import { useLocalization } from "@fluent/react";
+import {
+  DismissButton,
+  FocusScope,
+  mergeProps,
+  OverlayContainer,
+  useButton,
+  useDialog,
+  useModal,
+  useOverlay,
+  useOverlayPosition,
+  useOverlayTrigger,
+  VisuallyHidden,
+} from "react-aria";
+import { useOverlayTriggerState } from "react-stately";
+import styles from "./WhatsNewMenu.module.scss";
+import SizeLimitHero from "./images/size-limit-hero-10mb.svg";
+import SizeLimitIcon from "./images/size-limit-icon-10mb.svg";
+import SignBackInHero from "./images/sign-back-in-hero.svg";
+import SignBackInIcon from "./images/sign-back-in-icon.svg";
+import ForwardSomeHero from "./images/forward-some-hero.svg";
+import ForwardSomeIcon from "./images/forward-some-icon.svg";
+import { WhatsNewContent } from "./WhatsNewContent";
+import {
+  DismissalData,
+  useLocalDismissal,
+} from "../../../hooks/localDismissal";
+import { ProfileData } from "../../../hooks/api/profile";
+import { WhatsNewDashboard } from "./WhatsNewDashboard";
+import { useAddonData } from "../../../hooks/addon";
+import { isUsingFirefox } from "../../../functions/userAgent";
+
+export type WhatsNewEntry = {
+  title: string;
+  snippet: string;
+  content: ReactNode;
+  hero: string;
+  icon: string;
+  dismissal: DismissalData;
+  /**
+   * This is used to automatically archive entries of a certain age
+   */
+  announcementDate: {
+    year: number;
+    // Spelled out just to make sure it's clear we're not using 0-based months.
+    // Thanks, JavaScript...
+    month: 1 | 2 | 3 | 4 | 5 | 6 | 7 | 8 | 9 | 10 | 11 | 12;
+    date: number;
```

Comment:
Absolutely, renames are one of the things that TS makes really easy :) fd61ef2d 

To clarify why this was called `date`, that's to align with [`Date`'s `getDate` method](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/getDate) (which is called like that because [`getDay` returns the day of the week](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/getDay)). But that is indeed confusing.

Reply:
Yeah when I looked into that I noticed that `Date`'s [`UTC` method](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/UTC) (and some others) seem to use "day" - at least in the doc level. Thanks JavaScript.